packages/notifi-react-card/lib/components/intercom/ChatMessageSection.tsx

```diff
@@ -1,40 +1,96 @@
-import clsx from 'clsx';
 import React from 'react';
+import { Virtuoso } from 'react-virtuoso';
 
-import { ChatMessageDate, ChatMessageDateProps } from './ChatMessageData';
+import { useIntercomChat } from '../../hooks/useIntercomChat';
+import { ChatMessageDate, ChatMessageDateProps } from './ChatMessageDate';
+import {
+  ChatWindowIntroSection,
+  ChatWindowIntroSectionProps,
+} from './ChatWindowIntroSection';
 import { MessageList, MessageListProps } from './MessageList';
 
 export type ChatMessageSectionProps = Readonly<{
   classNames?: Readonly<{
-    container: string;
-    content: string;
-    date: ChatMessageDateProps['classNames'];
+    chatWindowIntro: ChatWindowIntroSectionProps['classNames'];
     messageList: MessageListProps['classNames'];
+    date: ChatMessageDateProps['classNames'];
   }>;
   chatMessageSectionIntroContent?: string;
 }>;
 
 export const ChatMessageSection: React.FC<ChatMessageSectionProps> = ({
   classNames,
-  chatMessageSectionIntroContent = 'What can we help you with today?',
+  chatMessageSectionIntroContent,
 }) => {
+  const {
+    conversation,
+    setIsScrolling,
+    setVisibleRange,
+    setAtTop,
+    isLoading,
+    hasNextPage,
+  } = useIntercomChat({
+    conversationId: '8c5ccf41d93c4bb9b5e1e1014002e923',
+  });
+
   return (
-    <div
-      className={clsx(
-        'NotifiIntercomChatMessageSection__container',
-        classNames?.container,
+    <>
+      {conversation.feed.length === 0 ? (
+        <ChatWindowIntroSection
+          classNames={classNames?.chatWindowIntro}
+          startDate={conversation.createdDate}
+          chatMessageSectionIntroContent={chatMessageSectionIntroContent}
+        />
+      ) : null}
+
+      {conversation.feed.length === 0 ? null : (
+        <Virtuoso
+          atTopStateChange={setAtTop}
+          className="virtual-container"
+          data={conversation.feed}
+          followOutput="auto"
+          isScrolling={setIsScrolling}
+          rangeChanged={setVisibleRange}
+          itemContent={(index, feed) => {
+            return (
+              <div key={index}>
+                {isLoading ? (
```

Comment:
Can you advise what is the intended use of the loading state? When should we see the loading state?

Reply:
Can you please implement a scroll seek? https://virtuoso.dev/scroll-seek-placeholders/

Reply:
We agreed that we will implement a scroll seek later. Hao has created a ticket in the backlog.